bluepysnap/spike_report.py

```diff
@@ -167,9 +167,16 @@ def report(self):
                 continue
             data = spikes.get(group=ids, t_start=self.t_start, t_stop=self.t_stop).to_frame()
             data["population"] = np.full(len(data), population)
-            res = pd.concat([res, data])
-        if not res.empty:
-            res["population"] = res["population"].astype("category")
+
+            if not data.empty:
+                dfs.append(data)
+
+        if len(dfs) == 0:
+            return pd.DataFrame()
```

Comment:
For consistency, it would be better to return an empty dataframe with the same schema (index and columns), instead of a generic empty dataframe, and add a test for it.
We fixed a similar issue for frame reports in https://github.com/BlueBrain/snap/pull/224